packages/flutter_fuchsia/lib/example_main.dart

```diff
@@ -0,0 +1,21 @@
+import 'dart:io';
+import 'dart:async';
+import 'dart:core';
+
+import 'package:logging/logging.dart';
+
+import 'flutter_fuchsia.dart';
+
+Future<Null> main(List<String> args) async {
+  // Sets up a basic logger to see what's happening.
+  Logger.root.level = Level.ALL;
+  Logger.root.onRecord.listen((LogRecord rec) {
+    print('[${rec.level.name}] -- ${rec.time}: ${rec.message}');
+  });
+  List<FlutterView> views =
+      await getFlutterViews('192.168.42.62', '../../', 'release-x86-64');
+  print(views);
+
+  // Program hangs here, so force an exit.
+  exit(0);
```

Comment:
This probably indicates a flaw in the API. You can use observatory to find any dangling "ports" (web-sockets maybe?). Perhaps the `FlutterView` class needs a `close` method to clean things up.

Reply:
Yeah, I'm a bit stumped on what causes this.  In the flutter_tools implementation they run through some shutdown hooks, then run a null completer after some timer expires, which calls the exit function explicitly.

I'm not sure if they have a similar problem, but I'll take a look at observatory.

Reply:
Fixed!